cinder/volume/drivers/vmware/volumeops.py

```diff
@@ -622,6 +631,54 @@ def extend_virtual_disk(self, requested_size_in_gb, path, dc_ref,
                  "%(size)s GB.",
                  {'path': path, 'size': requested_size_in_gb})
 
+    def extend_virtual_disk_online(self, requested_size_in_gb, path, vm_ref):
+        """Extend the virtual disk online to the requested size.
+
+        :param requested_size_in_gb: Size of the volume in GB
+        :param path: Datastore path of the virtual disk to extend
+        :param vm_ref: Reference to the VM instance
+        """
+        LOG.debug("Extending virtual disk: %(path)s to %(size)s GB.",
+                  {'path': path, 'size': requested_size_in_gb})
+        # VMWare API needs the capacity unit to be in KB, so convert the
+        # capacity unit from GB to KB.
+        cf = self._session.vim.client.factory
+        size_in_kb = requested_size_in_gb * units.Mi
+        config_spec = cf.create('ns0:VirtualMachineConfigSpec')
+        disk = None
+        devices = self._session.invoke_api(vim_util,
+                                           'get_object_property',
+                                           self._session.vim,
+                                           vm_ref,
+                                           'config.hardware.device')
+        for device in devices['VirtualDevice']:
+            if device.__class__.__name__ == "VirtualDisk" and \
+                    device.backing.fileName == path:
+                disk = device
```

Comment:
You could `break` here and then do the `else` from below in an `else` of the `for`.